node/src/components/storage.rs

```diff
@@ -475,6 +480,40 @@ impl Storage {
             metrics,
         };
 
+        if force_resync {
+            let force_resync_file_path = component.root_path().join(FORCE_RESYNC_FILE_NAME);
+            // Check if resync is already in progress. Force resync will kick
+            // in only when the marker file didn't exist before.
+            // Use `OpenOptions::create_new` to atomically check for the file
+            // presence and create it if necessary.
+            match OpenOptions::new()
+                .create_new(true)
+                .write(true)
+                .open(&force_resync_file_path)
+            {
+                Ok(_file) => {
+                    // When the force resync marker file was not present and
+                    // is now created, initialize force resync.
+                    info!("initializing force resync");
+                    // Default `storage.completed_blocks`.
+                    component.completed_blocks = Default::default();
+                    component.persist_completed_blocks()?;
```

Comment:
By creating the file first, we have a race condition where the attempt to persist the defaulted sequences could fail (node exits) and force-resync would hence not be re-attempted on the next run.

If we just check for the file's existence first, and only create the file once we successfully persist the defaulted sequences, we still have a race condition, but it's a little safer I think (another process could create the file after we check, but before we create it).

Probably not a big deal, both scenarios seem unlikely, and can be resolved by the operator if required.

Reply:
At this point, we're in the initialization phase of the storage component and the reactor itself. If something fails here (like `persist_completed_blocks`) and the node crashes, the operators have to manage the storage folder manually anyway. Given that this scenario is already highly unlikely, I'll leave it as is.